include/igl/surface_gradient.cpp

```diff
@@ -0,0 +1,25 @@
+// This file is part of libigl, a simple c++ geometry processing library.
+// 
+// Copyright (C) 2018 Zhongshi Jiang <jiangzs@nyu.edu>
+// 
+// This Source Code Form is subject to the terms of the Mozilla Public License 
+// v. 2.0. If a copy of the MPL was not distributed with this file, You can 
+// obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "surface_gradient.h"
+
+#include "grad.h"
+
+IGL_INLINE void igl::surface_gradient(const Eigen::MatrixXd &V, const Eigen::MatrixXi &F,
```

Comment:
break into a function that computes projection matrix and then client calls grad, build-projection, P*G